Remotion/Web/Development.WebTesting/DownloadInfrastructure/DownloadedFileFinder.cs

```diff
@@ -108,71 +107,45 @@ public FileInformationTuple (DateTime lastWriteTimeUtc, long length)
       {
         var newFiles = GetNewFiles (filesInDownloadDirectoryBeforeDownload, startedDownloadHandlingUtc);
 
-         //The first step, we haven't found a partial File yet to monitor the download process
-        if (partialFileState == null)
+        if (TryGetPartialFile (newFiles, out var partialFile)

```

Comment:
Some of the API choices on PartialFileState would make for a great refactoring exercise. Maybe this could be a Mob-session sometime?

Reply:
Would the starting point be the old design or the "simplified" version in this PR?

Reply:
I think we can start from here. The mainly because it looks like the refactoring helped and I don't want to spend a week in a mob training session.